actions/terra-k8s-linter/policy/deny.rego

```diff
@@ -0,0 +1,66 @@
+package main
+
+import data.kubernetes
+
+
+name = input.metadata.name
+
+required_deployment_labels {
+	input.metadata.labels.app.kubernetes.io/name
+	input.metadata.labels["app.kubernetes.io/instance"]
+	input.metadata.labels["app.kubernetes.io/version"]
+	input.metadata.labels["app.kubernetes.io/component"]
+	input.metadata.labels["app.kubernetes.io/part-of"]
+	input.metadata.labels["app.kubernetes.io/managed-by"]
+}
+
+required_probes {
+  input.spec.template.spec.containers["readinessProbe"]
+  input.spec.template.spec.containers["livenessProbe"]
+  input.spec.template.spec.containers["startupProbe"]
+}
+
+deny[msg] {
+	not required_deployment_labels
+	msg = sprintf("%s must include Kubernetes recommended labels", [name])
+}
+
+deny_liveness_probes[msg] {
+	not required_probes
+	msg = sprintf("%s Must have liveness probes", [name])
+}
+
+deny[msg] {
+  input.kind == "Deployment"
+  not input.spec.template.spec.securityContext.runAsNonRoot
+
+  msg := "Containers must not run as root"
+}
+
+deny_proper_service_account_name[msg] {
+  input.kind == "Deployment"
+  service_account_name := input.spec.template.spec.serviceAccountName
+  exp_service_account_name := sprintf("%s%s", [name, "-sa"])
+  not service_account_name == exp_service_account_name
+  msg := sprintf("Service account name expected: %s.  Service account name recieved: %s", [exp_service_account_name, service_account_name])
+}
+
+deny_proper_deployment_name[msg] {
+  input.kind == "Deployment"
+  not endswith(name, "-deployment")
+  msg := sprintf("Deployment name needs to end in '-deployment'. Current name: %s", [name])
+}
+
+deny_replicas_count[msg] {
+  input.kind == "Deployment"
+  input.spec.replicas < 3
+
+  msg := sprintf("Must have at least 3 replicas. %s has %b replicas.", [name, input.spec.replicas])
+}
+
+deny_revision_history[msg] {
+  input.kind == "Deployment"
+  input.spec.revisionHistoryLimit != 0
+
+  msg := "Deployments should set revisionHistoryLimit to 0"
+}
```

Comment:
This is a great point, @mflinn-broad! I actually pushed Eno in this direction because as part of the Helm design work I've been imagining a world in which Kubernetes manifests for Terra could be distributed across many repos. Even if we stick with terra-helmfile as a monorepo in the short term, I want to make sure that any new GitHub actions or other infrastructure we make can be adapted to support charts and values in application repos, too.

Maybe the answer here is:
* include a default set of Terra-standard policies with this action
* make it possible to disable the default policies with an input
* also make it possible to provide custom policies with an input

Reply:
sounds good to me. I'm fine with leaving it as is at least for a first draft if you've already discussed this.

Reply:
If we are going to define the linting rules centrally, maybe the action can start by logging a link to the where all the policies are defined

Reply:
Is this what you had in mind f2cf343?

Reply:
looks perfect, thanks Eno!